packages/core/ios/Sources/DdLogs.mm

```diff
@@ -0,0 +1,149 @@
+}
+
+RCT_REMAP_METHOD(error, withMessage:(NSString*)message
+                 withContext:(NSDictionary*)context
+                 withResolver:(RCTPromiseResolveBlock)resolve
+                 withRejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self error:message context:context resolve:resolve reject:reject];
+}
+
+RCT_REMAP_METHOD(debugWithError, withDebugMessage:(NSString*)message
+                 withErrorKind:(NSString*)errorKind
+                 withErrorMessage:(NSString*)errorMessage
+                 withStacktrace:(NSString*)stacktrace
+                 withContext:(NSDictionary*)context
+                 withResolver:(RCTPromiseResolveBlock)resolve
+                 withRejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self debugWithError:message errorKind:errorKind errorMessage:errorMessage stacktrace:stacktrace context:context resolve:resolve reject:reject];
+}
+
+RCT_REMAP_METHOD(infoWithError, withInfoMessage:(NSString*)message
+                 withErrorKind:(NSString*)errorKind
+                 withErrorMessage:(NSString*)errorMessage
+                 withStacktrace:(NSString*)stacktrace
+                 withContext:(NSDictionary*)context
+                 withResolver:(RCTPromiseResolveBlock)resolve
+                 withRejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self infoWithError:message errorKind:errorKind errorMessage:errorMessage stacktrace:stacktrace context:context resolve:resolve reject:reject];
+}            
+
+RCT_REMAP_METHOD(warnWithError, withWarnMessage:(NSString*)message
+                 withErrorKind:(NSString*)errorKind
+                 withErrorMessage:(NSString*)errorMessage
+                 withStacktrace:(NSString*)stacktrace
+                 withContext:(NSDictionary*)context
+                 withResolver:(RCTPromiseResolveBlock)resolve
+                 withRejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self warnWithError:message errorKind:errorKind errorMessage:errorMessage stacktrace:stacktrace context:context resolve:resolve reject:reject];
+}            
+
+RCT_REMAP_METHOD(errorWithError, withMessage:(NSString*)message
+                 withErrorKind:(NSString*)errorKind
+                 withErrorMessage:(NSString*)errorMessage
+                 withStacktrace:(NSString*)stacktrace
+                 withContext:(NSDictionary*)context
+                 withResolver:(RCTPromiseResolveBlock)resolve
+                 withRejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self errorWithError:message errorKind:errorKind errorMessage:errorMessage stacktrace:stacktrace context:context resolve:resolve reject:reject];
+}            
+
+// Thanks to this guard, we won't compile this code when we build for the old architecture.
+#ifdef RCT_NEW_ARCH_ENABLED
+- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:
+    (const facebook::react::ObjCTurboModule::InitParams &)params
+{
+    return std::make_shared<facebook::react::NativeDdLogsSpecJSI>(params);
```

Comment:
Does react automatically generate these `SpecJSI` structures?  Hence why they're in the `facebook::react` namespace?

Reply:
Exactly :)